cmd/nerdctl/container_prune.go

```diff
@@ -0,0 +1,103 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/nerdctl/pkg/containerinspector"
+	"github.com/containerd/nerdctl/pkg/namestore"
+	"github.com/spf13/cobra"
+)
+
+func newContainerPruneCommand() *cobra.Command {
+	containerPruneCommand := &cobra.Command{
+		Use:           "prune [flags]",
+		Short:         "Removes all stopped containers",
+		RunE:          containerPruneAction,
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+	containerPruneCommand.Flags().BoolP("force", "f", false, "Do not prompt for confirmation")
+	containerPruneCommand.Flags().String("filter", "", "not yet been implemented")
+	return containerPruneCommand
+}
+
+func containerPruneAction(cmd *cobra.Command, _ []string) error {
+	client, ctx, cancel, err := newClient(cmd)
+	if err != nil {
+		return err
+	}
+	defer cancel()
+	dataStore, err := getDataStore(cmd)
+	if err != nil {
+		return err
+	}
+	force, err := cmd.Flags().GetBool("force")
+	if err != nil {
+		return err
+	}
+	if !force {
+		var confirm string
+		fmt.Fprintf(cmd.OutOrStdout(), "%s", "WARNING! This will remove all stopped containers.\nAre you sure you want to continue? [y/N] ")
+		fmt.Fscanf(cmd.InOrStdin(), "%s", &confirm)
+
+		if strings.ToLower(confirm) != "y" {
+			return nil
+		}
+	}
+	ns, err := cmd.Flags().GetString("namespace")
+	if err != nil {
+		return err
+	}
+
+	ctx = namespaces.WithNamespace(ctx, ns)
+	var deletedContainers []string
+	containers, err := client.Containers(ctx)
```

Comment:
Can we filter the containers that need to be deleted?

Reply:
Filter on what basis?

Reply:
Let me work on it.

Reply:
I have added filters for volume and container according to docker